src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -534,8 +534,11 @@ ContainsMultipleDistributedRelations(PlannerRestrictionContext *
 	RelationRestrictionContext *restrictionContext =
 		plannerRestrictionContext->relationRestrictionContext;
 
-	uint32 referenceRelationCount = ReferenceRelationCount(restrictionContext);
-	uint32 totalRelationCount = list_length(restrictionContext->relationRestrictionList);
+	/* restrictionContext may contain duplicate entries */
+	uint32 referenceRelationCount = UniqueRelationCount(restrictionContext,
+														REFERENCE_TABLE);
+	uint32 totalRelationCount = UniqueRelationCount(restrictionContext,
+													ANY_CITUS_TABLE_TYPE);
 	uint32 nonReferenceRelationCount = totalRelationCount - referenceRelationCount;
```

Comment:
does the logic in this function still hold true in case of Citus local tables?

Reply:
Yes, thanks for spotting. We currently do not allow `CitusLocalTables` at this point, but still we should be careful. Simplified the code, only checking for `DISTRIBUTED_TABLE` as expected from the function.